pkg/x/http/transport.go

```diff
@@ -0,0 +1,80 @@
+package http
+
+import (
+	"cmp"
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/aquasecurity/trivy/pkg/version/app"
+)
+
+var (
+	defaultTransport = NewTransport(Options{})
+	mu               sync.RWMutex
+)
+
+type transportKey struct{}
+
+// WithTransport returns a new context with the given transport.
+// This is mainly for testing when a different HTTP transport needs to be used.
+func WithTransport(ctx context.Context, tr http.RoundTripper) context.Context {
+	return context.WithValue(ctx, transportKey{}, tr)
+}
+
+// Options configures the transport settings
+type Options struct {
+	Insecure  bool
+	Timeout   time.Duration
+	UserAgent string
+}
+
+// SetDefaultTransport sets the default transport configuration
+func SetDefaultTransport(tr http.RoundTripper) {
+	mu.Lock()
+	defer mu.Unlock()
+	defaultTransport = tr
+}
+
+// Transport returns the transport from the context, or the default transport if none is set.
+func Transport(ctx context.Context) http.RoundTripper {
+	t, ok := ctx.Value(transportKey{}).(http.RoundTripper)
+	if ok {
+		// If the transport is already set in the context, return it.
+		return t
+	}
+
+	mu.RLock()
+	defer mu.RUnlock()
+
+	return defaultTransport
+}
+
+// NewTransport creates a new HTTP transport with the specified options.
+// It should be used to initialize the default transport.
+// In most cases, you should use the `Transport` function to get the default transport.
+func NewTransport(opts Options) http.RoundTripper {
+	tr := http.DefaultTransport.(*http.Transport).Clone()
+
+	// Set timeout if specified
+	if opts.Timeout > 0 {
```

Comment:
I understand the desire to provide the flexibility to set  timeout but I've been bitten a few times using the Go native HTTP client which offers no timeout by default. Should we set it to a sane/reasonable value if no timeout is specified?

Reply:
That may be a good idea. What would be a reasonable default timeout? Since the default value of `--timeout` is 5 mins, should we use the same value?

Reply:
Yeah that makes sense to me.